third-party/rsutils/include/rsutils/concurrency/event.h

```diff
@@ -47,16 +47,19 @@ class event
 
     // Untrigger the event
     // Does not affect any threads
-    void clear()
+    // Returns the previous state
+    bool clear()
     {
         std::unique_lock< std::mutex > lock( _m );
+        bool was_set = _is_set;
         _is_set = false;
+        return was_set;
     }
 
     // Block until the event is set()
     // If already set, returns immediately
     // The event remains set when returning: it needs to be cleared...
-    void wait() const
+    void wait()
```

Comment:
Why did you remove the const? Semantically we don't expect `wait` to change internal event state

Reply:
Hmm I guess wait() waits for someone *else* to change it - even though you're expecting a change, you're no changing it yourself...
So I'll try putting back the const.